### ISSUE TITLE
Druid Postaggregations in superset: list index out of range

### ISSUE THREAD

**Issue by sowalassane** [apache/incubator-superset#5496]
Make sure these boxes are checked before submitting your issue - thank you!

- [ x] I have checked the superset logs for python stacktraces and included it here as text if any
- [ x] I have reproduced the issue with at least the latest released version of superset
- [ x] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
I am using superset version 0.26.3

### Expected results
Here is my  postaggregation metric

<img width="532" alt="capture d ecran 2018-07-26 a 11 45 01" src="https://user-images.githubusercontent.com/14936795/43255180-6c3673fa-90c9-11e8-8aeb-47f9ae5c0dde.png">
I expected . having a slice with the average i have computed with the postaggregation metric

### Actual results
List index out of range

**Comment by mistercrunch** [comment]
Please share the stack trace, and tag the folks are would touched these area last. You can find out who touched the failing method last using `git blame`.

**Comment by sowalassane** [comment]
Hi,
Thank you for your reply but i don't understand your explanation.
The stack trace is 'list index out of range'.
Regards

**Comment by ahsanshah** [comment]
Here is a sample postagg definition for an "Average" metric:

{
"type": "arithmetic",
"name": "div",
"fn": "/",
"fields": [
     {
          "fieldName": "sum__average_metric",
          "type": "fieldAccess",
          "name": "sum__average_metric"
     },
     {
               "type": "fieldAccess",
               "fieldName": "count",
               "name": "count"
          }
]
}

Are you defining the postagg properly?

**Comment by sowalassane** [comment]
Hi, thank you.

The context of my request:
- i have data from 2009 to 2014
- two measures nb

_personnes and quantity and two dimensions for exemple region and
department.

I want to calculate the average per region or per department.


Here is my definition of the average metric:


*{  "aggregations" : [    { "type" : "count", "name" : "rows" }, /* count
the total number of records */    { "type" : "longSum", "name" : "tot",
"fieldName" : "nb_personnes" }/* compute the total sum of my measure with
nb_personnesfield in my druid datasources */  ],/* query to compute the
average */  "postAggregations" : [{    "type"   : "arithmetic",    "name"
  : "average",    "fn"     : "/",    "fields" : [
      {"type":"fieldAccess", "name":"tot", "fieldName":"tot"},/* the total
sum that i calculated*/       { "type" : "fieldAccess", "name": "count"} /*
the total of records*/    ]  }]}Have i to specify an intervalle of time in
the definition ?*find in P.J respectively the stack trace when i use your
sample postagg definition and when i use my definition.



Le lun. 30 juil. 2018 à 05:15, ahsanshah <notifications@github.com> a
écrit :

**Comment by sowalassane** [comment]
Hi, thank you for your reply:
My definition for an "average" metric:
{
  "aggregations" : [
    { "type" : "count", "name" : "count" },
    { "type" : "longSum", "name" : "total", "fieldName" : "total_deces" }
  ],
  "postAggregations" : [{
    "type"   : "arithmetic",
    "name"   : "average",
    "fn"     : "/",
    "fields" : [
       {"type":"fieldAccess", "name":"average", "fieldName":"total"},
       { "type" : "fieldAccess", "name": "count"}
    ]
  }]
}
If i do that as you did what will i put on the "fieldName": "sum__average_metric" ?
The measure in druid column or i have to compute a sum before ?
Regards

**Comment by sowalassane** [comment]
Here is the error i get when i use respectively your sample of "average" metric and my definition 

<img width="1433" alt="capture d ecran 2018-07-30 a 10 16 46" src="https://user-images.githubusercontent.com/14936795/43401943-be6c80b0-9411-11e8-9b7e-7930c2daeaea.png">
<img width="1420" alt="capture d ecran 2018-07-30 a 10 42 21" src="https://user-images.githubusercontent.com/14936795/43401944-be9a8866-9411-11e8-91c7-57556febbc59.png">

**Comment by mistercrunch** [comment]
This is not directly related to this issue, but we're starting to recommend people use the Druid SQLAlchemy dialect and use that instead of the Druid connector. Post aggregations on that side of things are simple SQL expressions.

**Comment by sowalassane** [comment]
Thank you !
I will try that.

**Comment by mrw95** [comment]
"aggregations": [
    {
      "type": ""doubleSum"",
      "filter": {
        "type": "sum_facebook",
        "field": {
          "type": "selector",
          "dimension": "protocol_id",
          "value": total
        }
      },
 {
      "type": ""doubleSum",
      "name": "sum_facebook",
      "fieldNames": [
        "facebook"
      ],

    {
      "type": "doubleSum",
      "filter": {
        "type": "sum_youtube",
        "field": {
          "type": "selector",
          "dimension": "protocol_id",
          "value": total
        }
      },
 {
      "type": "doublesum",
      "name": "sum_youtube",
      "fieldNames": [
        "youtube"
      ],

"postAggregations" : [{
    "type"   : "arithmetic",
    "name"   : "protocol_id_average",
    "fn"     : "/",
    "fields" : [
       { "type"   : "arithmetic",
         "name"   : "",
         "fn"     : "/",
         "fields" : [
           { "type" : "fieldAccess",
             "name" : "protocol_id_average", 
             "fieldName" : "facebook" },
           { "type" : "fieldAccess", 
             "name" : "protocol_id_average", 
             "fieldName" : "youtube" }
         ]
       },
       { "type" : "constant", "name": "const", "value" : 100 }
    ]
  }]

this code right or wrong

**Comment by mrw95** [comment]
"aggregations": [
    {
      "type": ""doubleSum"",
      "filter": {
        "type": "sum_facebook",
        "field": {
          "type": "selector",
          "dimension": "protocol_id",
          "value": total
        }
      },
 {
      "type": ""doubleSum",
      "name": "sum_facebook",
      "fieldNames": [
        "facebook"
      ],

    {
      "type": "doubleSum",
      "filter": {
        "type": "sum_youtube",
        "field": {
          "type": "selector",
          "dimension": "protocol_id",
          "value": total
        }
      },
 {
      "type": "doublesum",
      "name": "sum_youtube",
      "fieldNames": [
        "youtube"
      ],

"postAggregations" : [{
    "type"   : "arithmetic",
    "name"   : "protocol_id_average",
    "fn"     : "/",
    "fields" : [
       { "type"   : "arithmetic",
         "name"   : "",
         "fn"     : "/",
         "fields" : [
           { "type" : "fieldAccess",
             "name" : "protocol_id_average", 
             "fieldName" : "facebook" },
           { "type" : "fieldAccess", 
             "name" : "protocol_id_average", 
             "fieldName" : "youtube" }
         ]
       },
       { "type" : "constant", "name": "const", "value" : 100 }
    ]
  }]

this code right or wrong
